test-refined/shared/src/main/scala/zio/test/refined/BooleanInstances.scala

```diff
@@ -16,8 +16,11 @@ trait BooleanInstances {
   ): DeriveGen[Refined[T, A Or B]] = {
     val genA: Gen[random.Random with Sized, T] = raGen.derive.map(_.value)
     val genB: Gen[random.Random with Sized, T] = rbGen.derive.map(_.value)
-    DeriveGen.instance(
-      Gen.oneOf[Random with Sized, T](genA, genB).map(Refined.unsafeApply)
-    )
+    DeriveGen.instance(orGen(genA, genB))
   }
+
+  def orGen[T, A, B](implicit
```

Comment:
When we compose generators like here we want to be polymorphic in the environment type. We can combine any two generators this way and the new generator will require all the services required by both generators, plus the `Random` service since we need to use random number generation to choose which generator to pull from each time. So I would rewrite this as:

```scala
  def orGen[R <: Random, T, A, B](implicit
    genA: Gen[R, T],
     genB: Gen[R, T]
   ): Gen[R, Refined[T, A Or B]] = Gen.oneOf[R, T](genA, genB).map(Refined.unsafeApply)
```